dags/socorro_import.py

```diff
@@ -18,39 +39,155 @@
+crash_report_parquet = SubDagOperator(
     task_id="crash_report_parquet",
-    job_name="Socorro Crash Reports Parquet",
-    execution_timeout=timedelta(hours=4),
-    instance_count=10,
-    env={"date": "{{ ds_nodash }}"},
-    uri="https://raw.githubusercontent.com/mozilla-services/data-pipeline/master/reports/socorro_import/ImportCrashData.ipynb",
-    output_visibility="public",
+    dag=dag,
+    subdag = moz_dataproc_pyspark_runner(
+        parent_dag_name=dag.dag_id,
+        dag_name='crash_report_parquet',
+        default_args=default_args,
+        cluster_name=cluster_name,
+        job_name="Socorro_Crash_Reports_to_Parquet",
+        # TODO - CI/CD for gcs and repo sync for artifact bucket jobs and gcp bootstrap folders
+        python_driver_code="gs://moz-fx-data-prod-airflow-artifacts/jobs/socorro_import_crash_data.py",
+        py_args=[
+            "-d", "{{ ds_nodash }}",
+            "--source-gcs-path", "gs://{}/v1/crash_report".format(gcs_data_bucket),
+            "--dest-gcs-path", "gs://{}/{}".format(gcs_data_bucket, dataset)
+        ],
+        idle_delete_ttl='14400',
+        num_workers=8,
+        worker_machine_type='n1-standard-8',
+        aws_conn_id=read_aws_conn_id,
+        gcp_conn_id=gcp_conn_id)
+)
+
+# TODO - change me later to remove the _2
+bq_gcp_conn_id= 'google_cloud_derived_datasets_2'
+bq_connection = GoogleCloudBaseHook(gcp_conn_id=bq_gcp_conn_id)
+gke_location="us-central1-a"
+gke_cluster_name="bq-load-gke-1"
+# TODO - change this to just s3://telemetry-parquet when ready
+dest_s3_key="s3://telemetry-parquet/socorro_crash_hwoo"
+
+# We don't use -m for parallel rsync because I ran into 503 while issues using it.
+gsutil_args = [
+    'gsutil', 'rsync', '-r',
+    'gs://{}/{}'.format(gcs_data_bucket, objects_prefix),
+    '{}/{}'.format(dest_s3_key, objects_prefix)
+]
+
+# Copy parquet data from gcs to s3
+gcs_to_s3 = GKEPodOperator(
+    task_id='gcs_to_s3',
+    gcp_conn_id=bq_gcp_conn_id,
+    project_id=bq_connection.project_id,
+    location=gke_location,
+    cluster_name=gke_cluster_name,
+    name='copy-parquet-data-back-to-s3',
+    namespace='default',
+    image="google/cloud-sdk:242.0.0-alpine",
+    arguments=gsutil_args,
+    env_vars={
+        "AWS_ACCESS_KEY_ID": "{{ var.value.socorro_access_key_id }}",
+        "AWS_SECRET_ACCESS_KEY": "{{ var.value.socorro_secret_access_key }}"
+    },
+    dag=dag,
+)
+
+# Not using load_to_bigquery since our source data is s3 json, instead of parquet
+# We do still use the parquet2bigquery container to load parquet into bq.
+# TODO _ change the dataset also to telemetry?
+bq_dataset='tmp'
+bq_table_name = '{}_{}'.format(dataset, dataset_version)
+
+docker_image='docker.io/mozilla/parquet2bigquery:20190722' # noqa
+
+gke_args = [
+    '-d', bq_dataset,
+    '-c', '10',
+    '-b', gcs_data_bucket,
+    '-R',
+    '-p', objects_prefix,
+    '--cluster-by', 'crash_date'
+]
+
+remove_bq_table_partition = BigQueryTableDeleteOperator(
```

Comment:
Why does the partition for the current date need to be removed? Does the `--glob-load` option do the right thing in this case if added to the `bq_load` step?

Reply:
Not sure, just following what we currently do for all of our other derived datasets.

Reply:
Can you add a comment to this affect? As someone not familiar with how we load derived datasets this also confused me.